app/services/selection.js

```diff
@@ -265,6 +282,7 @@ export default Service.extend({
           this.clearSelection();
           this.set('current', json);
         })
+        .then(() => {this.handleSelectedSummaryDataExplode.perform()});
```

Comment:
Just a note - I don't have a problem with this because this service already has this imperative `this.set` pattern throughout it but sure do with this was handled declaratively with some sort of "tracked" property.